docs/enhacement/20201211-nodepool_uniteddeployment.md

```diff
@@ -0,0 +1,374 @@
+    
+## Summary
+
+In the edge scenario, compute nodes have strong regional attributes. There may only be one worker node in the same physical location, and there may also be multiple worker nodes in the same physical location. Therefore, from the work node resource perspective of edge nodes, they need to be divided into different node pools (NodePool) to represent the same set of features. After dividing nodes pool, there will be a demand for application of grouping management, users need to be deployed application according to the nodepool, combined with the concept of nodepool, to deploy applications on different nodes in the pool, pool dimensions at the nodes to expansion of application, upgrade, such as operation, at the same time, network access is also carried out in accordance with the node pool dimensions of network communication.
+
+## Motivation
+
+- NodePool
+    Users can quickly learn which node pools or regions are in their cluster. You can also quickly see which nodes are in the node pool.
+    User can uniformly type label, annotation, and TAINts on nodes under the node pool.
+    This simplifies the operation and maintenance management of nodes.
+    
+- UnitedDeployment
+    User can use a template to deploy the application in a different node pool. This template can support one of Kubernetes deployment, Daemonset, Statefulset.
+    
+### Goals
+- Define the api of NodePool
+- Define the api of UnitedDeployment
+- Provide yurtunit-manager controller
+## Proposal
+
+- What is the plan for implementing this feature?
+
+  Provide yurtunit-manager operator to manage NodePool and UnitedDeployment crd
+
+### NodePool API
+
+``` go
+// NodePoolSpec defines the desired state of NodePool
+type NodePoolSpec struct {
+    // The type of the nodepool
+    // +optional
+    Type string `json:"type,omitempty"`
+
+    // A label query over nodes to consider for adding to the pool
+    // +optional
+    Selector *metav1.LabelSelector `json:"selector,omitempty"`
+
+    // If specified, the Labels will be added to all nodes.
+    // NOTE: existing labels with samy keys on the nodes will be overwritten.
+    // +optional
+    Labels map[string]string `json:"labels,omitempty"`
+
+    // If specified, the Annotations will be added to all nodes.
+    // NOTE: existing labels with samy keys on the nodes will be overwritten.
+    // +optional
+    Annotations map[string]string `json:"annotations,omitempty"`
+
+    // If specified, the Taints will be added to all nodes.
+    // +optional
+    Taints []v1.Taint `json:"taints,omitempty"`
+}
+
+// NodePoolStatus defines the observed state of NodePool
+type NodePoolStatus struct {
+    // Total number of ready nodes in the pool.
+    // +optional
+    ReadyNode int32 `json:"readyNode"`
```

Comment:
Could we change it to
ReadyNodeNum int32 `json:"readyNodeNum"`

Reply:
Yes. The `ReadyNodeNum` is more clear.